core/src/main/java/org/hibernate/ogm/dialect/spi/BaseGridDialect.java

```diff
@@ -44,4 +46,19 @@ public boolean supportsSequences() {
 	public DuplicateInsertPreventionStrategy getDuplicateInsertPreventionStrategy(EntityKeyMetadata entityKeyMetadata) {
 		return DuplicateInsertPreventionStrategy.LOOK_UP;
 	}
+
+	@Override
+	public void forEachTuple(ModelConsumer consumer, EntityKeyMetadata... entityKeyMetadatas) {
+		for (EntityKeyMetadata entityKeyMetadata : entityKeyMetadatas) {
+			List<Tuple> tuples = getTuples(entityKeyMetadata);
+			for (Tuple tuple : tuples) {
+				consumer.consume(tuple);
+			}
+		}
+	}
+
+	// Should be abstract, but high impact on all subclasses
+	protected List<Tuple> getTuples(EntityKeyMetadata entityKeyMetadata) {
```

Comment:
Do you think it's a good idea to encourage new developers to use this approach?

The method should at least be abstract, but then even so the fact that it returns a list of tuples tends to encourage loading all tuples in the JVM heap, which is quite prone to out of memory errors and also very inefficient.

I think you can use this approach on the ElasticSearch dialect while it's work in progress, but I wouldn't encourage this pattern in the parent `GridDialect`.

(besides you're also introducing some checkstyle errors by reformatting unrelated code areas, so better revert all changes to this class)